noir/Earthfile

```diff
@@ -58,16 +90,76 @@ packages:
       noir-repo/.github \
       noir-repo
 
-    COPY noir-repo/.github/scripts noir-repo/.github/scripts
-    COPY ./scripts/bootstrap_packages.sh ./scripts/bootstrap_packages.sh
+    ENV GIT_DIRTY=false
+    ENV GIT_COMMIT=$git_commit
+    ENV SOURCE_TIMESTAMP=$(date +%s)
+    ENV SOURCE_DATE_EPOCH=$(date +%s)
 
-    # TODO(AD) is this OK as a content hash?
-    ENV COMMIT_HASH=$(find . -type f -exec sha256sum {} ';' | sort | sha256sum | awk '{print $1}')
-    RUN echo CONTENT HASH $COMMIT_HASH | tee .content-hash
+    SAVE ARTIFACT noir-repo
+
+packages:
+    FROM +packages-deps
+
+    RUN echo CONTENT HASH $GIT_COMMIT | tee .content-hash
+    COPY ./scripts/bootstrap_packages.sh ./scripts/bootstrap_packages.sh
     RUN ./scripts/bootstrap_packages.sh
     SAVE ARTIFACT packages
     SAVE IMAGE --cache-hint
 
+packages-test-deps:
+  FROM +packages-deps
+
+  COPY +nargo/nargo /usr/src/noir/noir-repo/target/release/nargo
+  COPY +nargo/acvm /usr/src/noir/noir-repo/target/release/acvm
+
+  ENV NARGO_BACKEND_PATH=/usr/src/barretenberg/ts/dest/node/main.js
+  ENV PATH=$PATH:/usr/src/noir/noir-repo/target/release
+
+  WORKDIR /usr/src/barretenberg/ts
+  RUN yarn --immutable
+
+  WORKDIR /usr/src/noir/noir-repo
+  COPY --dir noir-repo/.github/scripts .github/scripts
+  RUN ./.github/scripts/wasm-bindgen-install.sh
+  RUN yarn build
+  RUN yarn workspace @noir-lang/noir_wasm run test:build_fixtures
+
+  SAVE ARTIFACT /usr/src /usr/src
+
+packages-test-node:
+  FROM +packages-test-deps
+  ENV NODE_OPTIONS=--max_old_space_size=8192
```

Comment:
This isn't actually needed now that `noir-lang/noir_js` is excluded (since it OOMs)